### PR TITLE
3sum and Cycled linked list problems

### PR DESCRIPTION
**Pull request by makcgr**

### CODE REVIEW

File: mgrachev/3Sum/Program.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+Given an integer array nums, return all the triplets [nums[i], nums[j], nums[k]] such that i != j, i != k, and j != k, and nums[i] + nums[j] + nums[k] == 0.
+
+Notice that the solution set must not contain duplicate triplets.
+
+------------
+
+Solution:
+
+Brute-force solution is to operate three pointers (three indices) and thus perform 3 loop iterations for the given array.
```
Comment (PShchegolevatykh) [inline mgrachev/3Sum/Program.cs:12]: As always good description. :)

File: mgrachev/linked list cycle/Program.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public bool HasCycle(ListNode head)
+    {
+        return CheckHasCycleRecursive(head, head);
+    }
+    public bool CheckHasCycleRecursive(ListNode current, ListNode head) {
+        if (current.Next == null)
+        {
+            return false;
+        }
+        return (current.Next == head ? true : CheckHasCycleRecursive(current.Next, head));
```
Comment (PShchegolevatykh) [inline mgrachev/linked list cycle/Program.cs:46]: Wow nice recursive version that we did not have before :)

**Review by PShchegolevatykh — APPROVED** [review]

File: mgrachev/3Sum/Program.cs (hunk 1 above)
Comment (makcgr) [reply mgrachev/3Sum/Program.cs:12]: Thanks, @PShchegolevatykh I appreciate it! Will do my best to keep it up!

File: mgrachev/linked list cycle/Program.cs (hunk 2 above)
Comment (makcgr) [reply mgrachev/linked list cycle/Program.cs:46]: I had a lot of experience with cycles in linked lists (hard to find bugs!) at my previous project (written in C almost fully)

**Pull request merged**